dbms/src/Common/TiFlashSecurity.h

```diff
@@ -100,11 +134,14 @@ struct TiFlashSecurityConfig
             {
                 redact_info_log = config.getBool("security.redact_info_log");
             }
+            return updated;
         }
+        return false;
     }
 
     void parseAllowedCN(String verify_cns)
     {
+        std::unique_lock lock(mu);
```

Comment:
Looks like when `parseAllowedCN()` is called inside `update()`, it will be deadlock?